### PR TITLE
rspq: implement RSP assertion support for overlays

### PR DESCRIPTION
**Pull request by rasky**
With this commit, rspq forwards to overlay's assert handler all assert
codes that it doesn't know of. This allows each overlays to handle their
own assertions.

### CODE REVIEW

File: src/rspq/rspq.c [hunk 1]
```diff
@@ -482,7 +485,13 @@ static void rspq_crash_handler(rsp_snapshot_t *state)
...
     printf("RSPQ: Current DRAM address: %08lx + GP=%lx = %08lx\n", 
         rspq->rspq_dram_addr, state->gpr[28], cur);
-    printf("RSPQ: Current Overlay: %02x\n", rspq->current_ovl / sizeof(rspq_overlay_t));
+
+    int ovl_idx = rspq->current_ovl / sizeof(rspq_overlay_t);
+    const char *ovl_name = "?";
+    if (ovl_idx < RSPQ_MAX_OVERLAY_COUNT && rspq_overlay_ucodes[ovl_idx])
+        ovl_name = rspq_overlay_ucodes[ovl_idx]->name;
+    printf("RSPQ: Current Overlay: %s (%02x)\n", ovl_name, ovl_idx);
+    printf("%p %s\n", rspq_overlay_ucodes[0], rspq_overlay_ucodes[ovl_idx]->name);
```
Comment (snacchus) [inline src/rspq/rspq.c]: This seems like a leftover from something IINM?

File: src/rspq/rspq.c [hunk 2]
```diff
@@ -506,15 +515,22 @@ static void rspq_crash_handler(rsp_snapshot_t *state)
 static void rspq_assert_invalid_command(rsp_snapshot_t *state)
 {
     rsp_queue_t *rspq = (rsp_queue_t*)state->dmem;
+
+    // Get overlay index a name. Be defensive against DMEM corruptions.
+    int ovl_idx = rspq->current_ovl / sizeof(rspq_overlay_t);
+    const char *ovl_name = "?";
+    if (ovl_idx < RSPQ_MAX_OVERLAY_COUNT && rspq_overlay_ucodes[ovl_idx])
+        ovl_name = rspq_overlay_ucodes[ovl_idx]->name;
```
Comment (snacchus) [inline src/rspq/rspq.c]: This could go into a helper function as it's a duplicate from the crash handler.

File: src/rspq/rspq.c [hunk 3]
```diff
@@ -506,15 +515,22 @@ static void rspq_crash_handler(rsp_snapshot_t *state)
...
+
     uint32_t dmem_buffer = RSPQ_DEBUG ? 0x140 : 0x100;
     uint32_t cur = dmem_buffer + state->gpr[28];
-    printf("Invalid command\nCommand %02x not found in overlay %02x\n", state->dmem[cur], rspq->current_ovl / sizeof(rspq_overlay_t));
+    printf("Invalid command\nCommand %02x not found in overlay %s (0x%01x)\n", state->dmem[cur], ovl_name, ovl_idx);
```
Comment (snacchus) [inline src/rspq/rspq.c:532]: Could add an additional check for internal commands here, so it says "builtin" or something instead of "?".

**Review by snacchus — CHANGES_REQUESTED** [review]
I tested it with rdpq and it works. The assert handler defined in the overlay ucode is correctly called. I Also found a couple of things that could be improved.

File: src/rspq/rspq.c (hunk 3 above)
Comment (snacchus) [reply src/rspq/rspq.c:532]: Actually since the command is not valid anyway maybe not.

File: src/rspq/rspq.c (hunk 1 above)
Comment (snacchus) [reply src/rspq/rspq.c]: Just referring to line 494 in case it's not clear.

Comment (rasky) [reply src/rspq/rspq.c]: Yes, removed.

File: src/rspq/rspq.c (hunk 2 above)
Comment (rasky) [reply src/rspq/rspq.c]: Done

File: src/rspq/rspq.c (hunk 3 above)
Comment (rasky) [reply src/rspq/rspq.c:532]: It still makes sense once it's moved into a separate function. Done.

**Pull request merged**